jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java

```diff
@@ -49,6 +49,16 @@
    */
   private static final Pattern portPattern = Pattern.compile("(\\d+)(?:/(tcp|udp))?");
 
+  /**
+   * Pattern used for parsing environment variables in the format {@code NAME=VALUE}. {@code NAME}
+   * should start with a letter and contain only letters (uppercase and lower) and numbers. {@code
+   * VALUE} can contain anything.
+   *
+   * <p>Example matches: NAME=VALUE, A12345=$$$$$
+   */
+  @VisibleForTesting
+  static final Pattern environmentPattern = Pattern.compile("([A-Za-z][A-Za-z0-9_]+)=(.*)");
```

Comment:
I believe environment variable names can actually contain any character (http://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html) - `These strings have the form name=value; names shall not contain the character '='.`, so we should probably have this pattern be like `(.*?)=(.*)`.

Reply:
[Here](https://github.com/opencontainers/runtime-spec/blob/master/config.md) it says:

> **env**: (array of strings, OPTIONAL) with the same semantics as IEEE Std 1003.1-2008's environ.

And in the link you shared:

"Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001 consist solely of uppercase letters, digits, and the '_' (underscore) from the characters defined in Portable Character Set and do not begin with a digit"

Reply:
Yep, for environment variables used in shell utilities (like `bash`), the names shouldn't consist of anything besides the alphanumerics, but for process environment variables, the definition is more lenient to allow any character besides `=`.

Reply:
reading is hard

Reply:
Shouldn't we make sure the name has at least one character though? I think what you gave will match something like "=ABC". Is that valid?

Reply:
That’s a good point - let’s change the star to a plus.

Reply:
That matches "====", I think `([^=].*?)=(.*)` should work

Reply:
Yea, I think having = in the value is okay